File: src/org/usfirst/frc5293/prefs/util/IntPref.java
package org.usfirst.frc5293.prefs.util;

import edu.wpi.first.wpilibj.Preferences;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class IntPref extends Pref<Integer> {
    public IntPref(String key, int defaultValue) {
        super(key, defaultValue);
    }

    @Override
    public Integer get() {
        return Preferences.getInstance().getInt(key, defaultValue);
    }

    @Override
    public void set(Integer value) {
        Preferences.getInstance().putInt(key, value);
    }
}

File: src/org/usfirst/frc5293/prefs/util/IntPref.kt
package org.usfirst.frc5293.prefs.util

import edu.wpi.first.wpilibj.Preferences

class IntPref(key: String, default: Int) : Pref<Int>(key, default) {
    override fun get() = Preferences.getInstance().getInt(key, default)

    override fun set(value: Int) = Preferences.getInstance().putInt(key, value)
}
